### PR TITLE
Added PrintFlags to ignore warnings

### PR DESCRIPTION
**Pull request by tpfau**
Added option to ignore warnings (e.g. for test cases, to reduce output)

**I hereby confirm that I have:**

- [X] Tested my code on my own machine
- [X] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [X] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [] with [X] to check the box)*

### CODE REVIEW

File: src/reconstruction/refinement/addMetabolite.m [hunk 1]
```diff
@@ -21,7 +21,9 @@
 %                   * Charge:        The Charge(s) (default NaN, int)
 %                   * b:             The accumulation(s) or release(s) (default 0, double)
 %                   * csense:        The sense of this metabolite (default 'E', char)
-%
+%                   * printLevel - 1  : normal output
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/addMetabolite.m]: `printLevel = 0` is mute.  May you change the documentation here?

File: src/reconstruction/refinement/addMetabolite.m [hunk 2]
```diff
@@ -143,7 +147,9 @@
             cmetName = metName{i};
             if strcmp(cmetName,'')
                 model.metNames{end+1,1} = regexprep(cmetID,'(\[.+\]) | (\(.+\))','') ;
-                warning(['Metabolite name for ' metID{i} ' set to ' model.metNames{end}]);
+                if printLevel >= 0
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/addMetabolite.m]: `if printLevel > 0`

File: src/reconstruction/refinement/addReaction.m [hunk 3]
```diff
@@ -176,6 +177,12 @@
 revFlag = parser.Results.reversible;
 geneNameList = parser.Results.geneNameList;
 systNameList = parser.Results.systNameList;
+if oldRxnFlag
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/addReaction.m]: `if oldRxnFlag && printLevel > 0`

File: src/reconstruction/refinement/addReaction.m [hunk 4]
```diff
@@ -254,7 +261,9 @@
 %Give warning and combine the coeffeicient if a metabolite appears more than once
 [metaboliteListUnique,~,IC] = unique(metaboliteList);
 if numel(metaboliteListUnique) ~= numel(metaboliteList)
-    warning('Repeated mets in the formula for rxn ''%s''. Combine the stoichiometry.', rxnID)
+    if printLevel >= 0
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/addReaction.m]: `if printLevel > 0`

File: src/reconstruction/refinement/addReaction.m [hunk 5]
```diff
@@ -281,8 +290,10 @@
     if (isInModel(i))
         Scolumn(metID(i),1) = stoichCoeffList(i);
     else
-        warning(['Metabolite ' metaboliteList{i} ' not in model - added to the model']);
-        model = addMetabolite(model,metaboliteList{i},metaboliteList{i});
+        if printLevel >= 0
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/addReaction.m]: `if printLevel > 0`

File: src/reconstruction/refinement/addReaction.m [hunk 6]
```diff
@@ -316,7 +327,9 @@
 %If the reaction is already present, and the updated reaction is not one of
 %the reactions, which actually is matching.
 if any(~(duplicatePos == rxnPos))
-    warning(['Model already has the same reaction you tried to add: ' modelOrig.rxns{rxnIDexists}]);
+    if printLevel >= 0
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/addReaction.m]: `if printLevel > 0`

**Comment by tpfau** [comment]
Hmm.... Mute should not throw warnings? 
These warnings are not just "normal" output, but indicators that something might have gone wrong in these functions.

**Comment by laurentheirendt** [comment]
If that is the case, why is then a `warning` dependent on `printLevel`?

**Comment by tpfau** [comment]
Point taken.
I'll close this.

**Pull request closed without merging**